crates/types/src/types.rs

```diff
@@ -102,34 +108,54 @@ pub struct ComputeUnitsEstimationResult {
 }
 
 /// The struct for storing the profiling results.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct KeyedProfileResult {
+    pub slot: u64,
+    pub key: UuidOrSignature,
+    pub profile: TransactionProfileResult,
+}
+
+// Tx with ixs [A, B, C]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct TransactionProfileResult {
+    pub instruction_profiles: Option<ProfiledInstructionsMap>, // Contains { [A] : {..result}, [A+B]: {..result} }
```

Comment:
Can we pull the `indexmap` crate and use an `IndexMap` here?
This would allow us to get ordered map, which seems important in this context.